addons/osfstorage/models.py

```diff
@@ -128,17 +128,27 @@ def path(self):
     def is_checked_out(self):
         return self.checkout is not None
 
-    def delete(self, user=None, parent=None, **kwargs):
-        if self.node.preprint_file and self.node.preprint_file.pk == self.pk:
-            self.node._is_preprint_orphan = True
-            self.node.save()
+    @property
+    def _delete_allowed(self):
```

Comment:
Minor: It feels a bit unintuitive for a property to do validation and raise validation errors. Perhaps make this a method, e.g. `_check_delete_allowed()`. I don't feel strongly about it though.